Source/ARTPush.m

```diff
@@ -54,6 +55,16 @@ - (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
     [_logger error:@"ARTPush: device token not received (%@)", [error localizedDescription]];
 }
 
++ (ARTPushActivationStateMachine *)activationMachine {
+    static dispatch_once_t once;
+    static id activationMachineInstance;
+    dispatch_once(&once, ^{
+        // Error: ARTPush.m:62:81: Instance variable '_httpExecutor' accessed in class method
+        //activationMachineInstance = [[ARTPushActivationStateMachine alloc] init:_httpExecutor];
+    });
```

Comment:
@tcard `ARTPushActivationStateMachine` should be a singleton right? I need to have in mind which data the state machine needs to perform is work. Does it need more than the local device?